akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -40,7 +46,13 @@ sealed abstract class OutPort { self: Outlet[_] ⇒
   /**
    * INTERNAL API
    */
-  private[stream] var id: Int = -1
+  @volatile private[stream] var id: Int = -1
+
+  /**
+   * INTERNAL API
+   */
+  @volatile private[stream] var mappedTo: OutPort = this
```

Comment:
was the missing volatile a bug previously, or is the new traversal different in this perspective?